src/metabase/api/segment.clj

```diff
@@ -25,6 +26,12 @@
   (check-superuser)
   (check-404 (segment/retrieve-segment id)))
 
+(defendpoint GET "/"
+  "Fetch *all* `Segments`."
+  []
+  (-> (db/select Segment, :is_active true)
+      (hydrate :creator)))
```

Comment:
Does this fetch all creators in a single SELECT or are we fetching each one separately?